src/App.js

```diff
@@ -1,11 +1,124 @@
 import React, { Component } from 'react';
-import './App.css';
+import axios from 'axios';
+import MovieList from './MovieList';
+import SearchBar from './SearchBar';
 
 class App extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      movies: [],
+      savedMovies: [],
+      searchText: ''
+    };
+  }
+  componentDidMount() {
+    this.getPopularMovies();
+    this.getSavedMovies();
+  }
+  getSavedMovies() {
+    axios.get('http://localhost:4000/movies')
+      .then(resp => {
+        this.setState({
+          savedMovies: resp.data
+        });
+      });
+  }
+  getPopularMovies() {
+    const savedMoviesIdArr = this.state.savedMovies.map(savedMovie => savedMovie.id);
+    axios.get('http://api.themoviedb.org/3/movie/popular?api_key=2dba200e2682e0f8903ed87b9c9e02d1')
+      .then(resp => {
+        // Only return popular movies whos IDs are NOT found in my savedMovies
+        this.setState({
+          movies: resp.data.results.filter(resultsMovie =>
+            !savedMoviesIdArr[savedMoviesIdArr.indexOf(resultsMovie.id)])
+        });
+      });
+  }
+  handleSearchBarChange(event) {
+    this.setState({
+      searchText: event.target.value
+    });
+  }
+  handleSubmit(event) {
+    event.preventDefault();
+    const savedMoviesIdArr = this.state.savedMovies.map(savedMovie => savedMovie.id);
+    axios.get(`https://api.themoviedb.org/3/search/movie?api_key=2dba200e2682e0f8903ed87b9c9e02d1&language=en-US&query=${this.state.searchText}&page=1&include_adult=false`)
+      .then(resp => {
+        this.setState({
+          // Only return searched movies whos IDs are NOT found in my savedMovies
+          movies: resp.data.results.filter(resultsMovie =>
+            !savedMoviesIdArr[savedMoviesIdArr.indexOf(resultsMovie.id)])
```

Comment:
Same here, also, you are have this idea a couple of times about checking to see if a movie is in your `savedMovies` array. When we see repetition like this it is usually a good sign that we can move some of that logic into a helper function.

This makes our code cleaner, and then if you ever have any changes as to how you are doing this type of check, you only need to update in one place.